Multisig/App/AppDelegate.swift

```diff
@@ -25,6 +25,8 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         Tracker.shared.setSafeCount(Safe.count)
         Tracker.shared.setNumKeysImported(SafeTransactionSigner.numberOfKeysImported())
 
+        Safe.updateCachedNames()
```

Comment:
Does it increase app loading time when launched on the device?

Reply:
I think not much. I could do it on background thread as well

Reply:
But I remember I had some issues when working with db queries on different threads. This might require a bit more effort.

Reply:
afaik our core data must be done on the main thread :/

Reply:
At the moment yes - otherwise the app will crash as we use one context. But there is a possibility to work on different threads. I created #915 issue